cpp/test/sg/rf_treelite_test.cu

```diff
@@ -320,10 +308,11 @@ class RfTreeliteTestReg : public RfTreeliteTestCommon<T, L> {
     // #class for multi-class classification
     this->task_category = 1;
 
-    // Populate labels randomly.
-    for (int i = 0; i < this->params.n_inference_rows; i++) {
-      this->labels_h.push_back(static_cast<float>(rand()) /
-                               (static_cast<float>(RAND_MAX / 10.0)));
+    for (int i = 0; i < this->params.n_rows; i++) {
+      // Generate labels which are linear to #row plus small noise.
+      this->labels_h.push_back(
+        i + static_cast<float>(rand()) /
```

Comment:
For the test case, labels should be linear on the data row, not its index.

Suppose that `x_1`, ..., `x_m` is a data row. Then a good label for regression is `y = a_1 * x_1 + ... + a_m * x_m + noise`, where `a_1`, ..., `a_m` are some coefficients (maybe random) that are the same for all rows. If there are NaNs among the input data, feel free to ignore them when summing. If all rows are viewed as a matrix, then this is a matrix vector multiplication followed by adding a random noise vector.

This can be easily done on the GPU.

For classification, labels as for regression may be used followed by applying a threshold.